### PR TITLE
Multiple locale paths support.

### PR DESCRIPTION
**Pull request by huacnlee**
We may need multiple locale paths in sometimes.

Example:

I write a submodule for Koa, need locale file.
User use the module, need rewrite the locale file with itself or add more locales.

### CODE REVIEW

**Comment by hotoo** [comment]
Why you need this feature, and how to resolve locales conflict?

File: index.js [hunk 1]
```diff
@@ -27,26 +28,45 @@ module.exports = function (app, options) {
...
 
       if (name.endsWith('.js') || name.endsWith('.json')) {
-        resources[locale] = require(filepath);
+        resource = require(filepath);
       } else if (name.endsWith('.properties')) {
-        resources[locale] = ini.parse(fs.readFileSync(filepath, 'utf8'));
+        resource = ini.parse(fs.readFileSync(filepath, 'utf8'));
       }
+
+      resources[locale] = resources[locale] || [];
```
Comment (fengmk2) [inline index.js]: Should be `{}`, not `[]`

File: index.js [hunk 2]
```diff
@@ -27,26 +28,45 @@ module.exports = function (app, options) {
...
   var resources = {};
 
-  if (fs.existsSync(localeDir)) {
-    var names = fs.readdirSync(localeDir);
-    for (var i = 0; i < names.length; i++) {
-      var name = names[i];
-      var filepath = path.join(localeDir, name);
+  if (localeDir) {
+    if (localeDirs.indexOf(localeDir) === -1) {
+      localeDirs.push(localeDir);
```
Comment (fengmk2) [inline index.js]: Need a test case to coverage this logic.

File: History.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+1.3.0 / 2015-09-14
+==================
+
+ * feat: Multiple locale paths support.
```
Comment (fengmk2) [inline History.md]: No need to change history, I will auto update histroy when npm publish run.

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
@huacnlee koa-locales@1.3.0

File: package.json [hunk 4]
```diff
@@ -30,7 +30,8 @@
     "mm": "~1.1.0",
     "mocha": "*",
     "pedding": "~1.0.0",
-    "supertest": "~1.0.1"
+    "supertest": "~1.0.1",
+    "merge-descriptors": "~1.0.0"
```
Comment (fengmk2) [inline package.json]: ... should be in dependencies

Comment (huacnlee) [inline package.json]: Ooops...

**Pull request merged**